bundles/org.openhab.binding.plex/src/main/java/org/openhab/binding/plex/internal/handler/PlexApiConnector.java

```diff
@@ -198,28 +204,60 @@ public boolean getApi() {
                 }
             }
             return false;
-        } catch (IOException e) {
+        } catch (IOException | InterruptedException | TimeoutException | ExecutionException e) {
             logger.debug("An exception occurred while fetching API :'{}'", e.getMessage(), e);
         }
         return false;
     }
 
     /**
-     * Make an HTTP request and return the class object that was used when calling.
+     * Make an HTTP request and return the response as a string.
      *
-     * @param <T> Class being used(dto)
      * @param method GET/POST
      * @param url What URL to call
      * @param headers Additional headers that will be used
-     * @param type class type for the XML parsing
-     * @return Returns a class object from the data returned by the call
+     * @param verify Flag to indicate if ssl certificate checking should be done
+     * @return Returns a string of the http response
      * @throws IOException
+     * @throws ExecutionException
+     * @throws TimeoutException
+     * @throws InterruptedException
      */
-    private <T> T doHttpRequest(String method, String url, Properties headers, Class<T> type) throws IOException {
-        String response = HttpUtil.executeUrl(method, url, headers, null, null, REQUEST_TIMEOUT_MS);
+    private String doHttpRequest(String method, String url, Properties headers, boolean verify)
+            throws IOException, InterruptedException, TimeoutException, ExecutionException {
+        final String response;
+        if (verify) {
+            // Requests sent to the PLEX servers should use certificate checking
+            response = HttpUtil.executeUrl(method, url, headers, null, null, REQUEST_TIMEOUT_MS);
+        } else {
+            // Requests sent to the local server need to bypass certificate checking via the custom httpClient
+            final Request request = httpClient.newRequest(url).method(HttpUtil.createHttpMethod(method));
+            if (headers != null) {
+                for (String httpHeaderKey : headers.stringPropertyNames()) {
+                    if (httpHeaderKey.equalsIgnoreCase(HttpHeader.USER_AGENT.toString())) {
+                        request.agent(headers.getProperty(httpHeaderKey));
+                    } else {
+                        request.header(httpHeaderKey, headers.getProperty(httpHeaderKey));
+                    }
+                }
+            }
```

Comment:
The core `HttpUtil` code is partly like that because of a bugfix because of the way Jetty works - see openhab/openhab-core#3411. The other part of the story is that the method was bound by contract to have agent delivered inside `Properties`. We are not bound by this contract, but can freely write the code.

I think it would be better to avoid **for each call** to put 9-10 headers into a `HashMap`, with `User-Agent` being one of them and then iterate through the map in order to apply it for building a `Request`. while taking special care of `User-Agent` by string comparison. At least the `User-Agent` handling part seems like a work-around as it could have been avoided in the first place.

But I can appreciate that you don't want to mess too much with this code since you are not using the binding yourself, so let's leave it as is for now.